lib/plsql/jdbc_connection.rb

```diff
@@ -348,7 +348,7 @@ def plsql_to_ruby_data_type(metadata)
         [Java::OracleSql::CLOB, nil]
       when "BLOB"
         [Java::OracleSql::BLOB, nil]
-      when "NUMBER"
+      when "NUMBER", "BINARY_FLOAT", "BINARY_DOUBLE"
```

Comment:
> I understand that the goal would be to have mathematical operations outcomes aligned between Ruby and Oracle for those datatypes so that:
> 2/3 in ruby using Float = 2/3 in Oracle using BINARY_DOUBLE and
> 2/3 in ruby using single-Float = 2/3 in Oracle using BINARY_FLOAT.
> Would that be a good expectation?

I suspect the above expectation might be too strong, even though possibly a good approximation. Maybe the built-in Ruby `Float` would be enough: the main goal is to avoid loss of precision just because of decimal <-> binary conversions; and I guess conversion between 64 and 32 bit binary floating point formats would be with less side effects.
(Too strong since Ruby and Oracle may be running on different hardware architectures. Or even on same architecture I'm not sure it's working 100% the same way).

Another test may be an identity pl/sql function (returning its argument unmodified): and see if we have any difference between the input and output when we run it from Ruby.